proposals/flexible-vectors/FlexibleVectors.md

```diff
@@ -141,8 +141,53 @@ def S.replace_lane(a, i, x):
 The input lane value, `x`, is interpreted the same way as for the splat
 instructions. For the `i8` and `i16` lanes, the high bits of `x` are ignored.
 
+### Shuffles
 
-### Integer arithmetic
+#### Left lane-wise shift by scalar
+
+* `vec.i8.lane_shift_left(a: vec.i8, x: i32) -> vec.i8`
+* `vec.i16.lane_shift_left(a: vec.i16, x: i32) -> vec.i16`
+* `vec.i32.lane_shift_left(a: vec.i32, x: i32) -> vec.i32`
+* `vec.i64.lane_shift_left(a: vec.i64, x: i32) -> vec.i64`
```

Comment:
Random drive-by comment: you don't want to use the same nomenclature as the 128-bit spec? e.g. `shl/shr_u/shr_s`?

Reply:
Those are already in - this change introduces "packed shift" instructions (for example, `psll*`/`psrl*` instructions on x86 and `lsl`/`lsr` on Arm). I do think we need a better name here though 😄

Reply:
Oh, it's shifting lanes, not bits!

> for example, psll*/psrl* instructions on x86 and lsl/lsr on Arm

But `shl` already uses `PSLL{W/D/Q}`, e.g., so do you mean that this would use like `PSLLDQ`/`VPSLLDQ`?

Reply:
:+1: for a more clear name 😆